encoding/gproperties/gproperties.go

```diff
@@ -0,0 +1,137 @@
+// Copyright GoFrame Author(https://goframe.org). All Rights Reserved.
+//
+// This Source Code Form is subject to the terms of the MIT License.
+// If a copy of the MIT was not distributed with this file,
+// You can obtain one at https://github.com/gogf/gf.
+
+// Package gproperties provides accessing and converting for .properties content.
+package gproperties
+
+import (
+	"bytes"
+	"sort"
+	"strings"
+
+	"github.com/gogf/gf/v2/errors/gerror"
+	"github.com/gogf/gf/v2/internal/json"
+	"github.com/magiconair/properties"
+	"github.com/spf13/cast"
+)
+
+// Decode converts properties format to map.
+func Decode(data []byte) (res map[string]interface{}, err error) {
+	res = make(map[string]interface{})
+	pr, err := properties.Load(data, properties.UTF8)
+	if err != nil || pr == nil {
+		err = gerror.Wrapf(err, `Lib magiconair load Properties data failed.`)
+		return nil, err
+	}
+	for _, key := range pr.Keys() {
+		// ignore existence check: we know it's there
+		value, _ := pr.Get(key)
+		// recursively build nested maps
+		path := strings.Split(key, ".")
+		lastKey := strings.ToLower(path[len(path)-1])
+		deepestMap := deepSearch(res, path[0:len(path)-1])
+
+		// set innermost value
+		deepestMap[lastKey] = value
+	}
+	return res, nil
+}
+
+// Encode converts map to properties format.
+func Encode(data map[string]interface{}) (res []byte, err error) {
+	pr := properties.NewProperties()
+
+	flattened := map[string]interface{}{}
+
+	flattened = flattenAndMergeMap(flattened, data, "", ".")
+
+	keys := make([]string, 0, len(flattened))
+
+	for key := range flattened {
+		keys = append(keys, key)
+	}
+
+	sort.Strings(keys)
+
+	for _, key := range keys {
+		_, _, err := pr.Set(key, cast.ToString(flattened[key]))
```

Comment:
Use `gconv.String` instead of `cast.ToString`.

Reply:
Use gconv.String instead of cast.ToString.